src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -298,10 +298,10 @@ internal ValueTask<FlushResult> FlushAsync(CancellationToken cancellationToken)
             {
                 CommitUnsynchronized();
 
-                _readerAwaitable.Complete(out completionData);
-
                 cancellationTokenRegistration = _writerAwaitable.AttachToken(cancellationToken, s_signalWriterAwaitable, this);
```

Comment:
Why are we ever disposing this registration below before the returned ValueTask completes?

Reply:
Because that's the old registration we replaced.